Add Data Access Control

urseberry opened the pull request:
<!-- *Note: Please remember to review the Datadog Documentation [Contribution Guidelines](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md) if you have not yet done so.* -->

### What does this PR do? What is the motivation?
<!-- A brief description of the change being made with this pull request. What is your motivation for the PR? -->

### Merge instructions

<!-- If you're waiting for a release or there are other considerations that you want us to be aware of, list them here. -->

Merge queue is enabled in this repo. To have it automatically merged after it receives the required reviews, create the PR (from a branch that follows the `<yourname>/description` naming convention) and then add the following PR comment:

```
/merge
```

### Additional notes
<!-- Anything else we should know when reviewing?-->

<!-- Previewing the PR: Assuming you are a Datadog employee and named your branch `<yourname>/<description>`, a preview build will run and links to the preview output will be auto-generated and posted in the PR comments. The links will 404 until the preview build is finished running. -->

aliciascott commented on content/en/account_management/rbac/data_access.md:
```diff
@@ -0,0 +1,183 @@
+
+## Select tags for access
+
+Each restricted dataset can control access to multiple types of data, such as metrics. You are free to use the same or different tags across multiple types of telemetry. Within each telemetry type, you must use a _single_ tag or attribute to define your access strategy.
```
```suggestion
Each Restricted Dataset can control access to multiple types of data, such as metrics. You are free to use the same or different tags across multiple types of telemetry. Within each telemetry type, you must use a _single_ tag or attribute to define your access strategy.
```

aliciascott commented on content/en/account_management/rbac/data_access.md:
```diff
@@ -0,0 +1,183 @@
+Each restricted dataset can control access to multiple types of data, such as metrics. You are free to use the same or different tags across multiple types of telemetry. Within each telemetry type, you must use a _single_ tag or attribute to define your access strategy.
+
+If you have too many combinations of tags or attributes to fit within these constraints, consider [revisiting your tagging][4] to define a new tag that better reflects your access strategy.
+
```
```suggestion
# Supported examples and restrictions for Restricted Datasets
```
I think the flow and formatting can be improved here but feel free to take or reject the suggestion

aliciascott commented on content/en/account_management/rbac/data_access.md:
```diff
@@ -0,0 +1,183 @@
+
+If you have too many combinations of tags or attributes to fit within these constraints, consider [revisiting your tagging][4] to define a new tag that better reflects your access strategy.
+
+**Supported example**
```
```suggestion
## Supported example
```

aliciascott commented on content/en/account_management/rbac/data_access.md:
```diff
@@ -0,0 +1,183 @@
+
+**Supported example**
+
+Restricted dataset 1: 
```
```suggestion
### Restricted dataset 1
```

aliciascott commented on content/en/account_management/rbac/data_access.md:
```diff
@@ -0,0 +1,183 @@
+**Supported example**
+
+Restricted dataset 1: 
+* Telemetry type: RUM 
```
```suggestion
- **Telemetry Type**: RUM  
   - **Filters**: `@application.id:ABCD`
```

aliciascott commented on content/en/account_management/rbac/data_access.md:
```diff
@@ -0,0 +1,183 @@
+
+Restricted dataset 1: 
+* Telemetry type: RUM 
+    * Filters: `@application.id:ABCD`
```
```suggestion
```

aliciascott commented on content/en/account_management/rbac/data_access.md:
```diff
@@ -0,0 +1,183 @@
+* Telemetry type: RUM 
+    * Filters: `@application.id:ABCD`
+
+Restricted dataset 2: 
```
```suggestion
### Restricted dataset 2
```

aliciascott commented on content/en/account_management/rbac/data_access.md:
```diff
@@ -0,0 +1,183 @@
+* Telemetry type: Metrics
+    * Filters: `env:prod`
+
+**Not supported**
```
```suggestion
## Not supported examples
```

aliciascott commented on content/en/account_management/rbac/data_access.md:
```diff
@@ -0,0 +1,183 @@
+* Telemetry type: RUM
+    * Filters: `@application.id:IJKL` `env:prod`
+
+While this example uses the right `@application.id` tag for RUM as was done for restricted dataset 1, the limit is 1 tag per telemetry type. Instead, consider either creating a restricted dataset with _either_ `application.id` or `env`, or identifying a different tag that better combines these attributes.
```
```suggestion
This example correctly uses the `@application.id` tag for RUM, as was done for restricted dataset 1. However, the limit is one tag per telemetry type. Instead, consider creating a restricted dataset with _either_ `application.id` or `env`, or identify a different tag that better combines these attributes.
```

aliciascott commented on content/en/account_management/rbac/data_access.md:
```diff
@@ -0,0 +1,183 @@
+
+**Not supported**
+
+Restricted dataset 1: 
```
```suggestion
### Restricted dataset 1: 
```

aliciascott commented on content/en/account_management/rbac/data_access.md:
```diff
@@ -0,0 +1,183 @@
+* Telemetry type: RUM 
+    * Filters: `@application.id:ABCD`
+
+Restricted dataset 2:
```
```suggestion
### Restricted dataset 2:
```

aliciascott commented on content/en/account_management/rbac/data_access.md:
```diff
@@ -0,0 +1,183 @@
+
+Restricted dataset 1 uses `@application.id` as the tag for RUM data, so a new restricted dataset can't change to a different tag. Instead, consider reconfiguring restricted dataset 2 to use `@application.id`, or changing all of your restricted datasets with RUM data to use another tag.
+
+**Not supported**
```
```suggestion
## Not supported examples
```

aliciascott commented on content/en/account_management/rbac/data_access.md:
```diff
@@ -0,0 +1,183 @@
+
+**Not supported**
+
+Restricted dataset 1: 
```
```suggestion
### Restricted dataset 1: 
```

aliciascott commented on content/en/account_management/rbac/data_access.md:
```diff
@@ -0,0 +1,183 @@
+* Telemetry type: RUM 
+    * Filters: `@application.id:ABCD`
+
+Restricted Dataset 2: 
```
```suggestion
### Restricted Dataset 2: 
```

aliciascott commented on content/en/account_management/rbac/data_access.md:
```diff
@@ -0,0 +1,183 @@
+
+## Access enforcement
+
+Users in a Datadog organization with Data Access Control enabled can only see query results for data to which they have access, such as in a Dashboard, in an Explorer, or through the API. A restricted dataset removes access from the data bound in the restricted dataset from users who are not permitted across all Datadog experiences and entry points.
```
```suggestion
Users in a Datadog organization with Data Access Control enabled can only see query results for data to which they have access, such as in a Dashboard, in an Explorer, or through the API. A restricted dataset removes access to the data defined in the restricted dataset for users who are not permitted, across all Datadog experiences and entry points.
```

aliciascott commented on content/en/account_management/rbac/data_access.md:
```diff
@@ -0,0 +1,183 @@
+
+### Data explorers
+
+When exploring Datadog with restrictions enabled, users without permissions are still able to cycle through the list of asset names (for example: applications or metrics), but they do not see query results, top tags, or facet details that have been restricted by restricted datasets. When querying a metric, for example, that contains data which is restricted from the user, a blank graph returns. I ...
```
```suggestion
When exploring Datadog with restrictions enabled, users without permissions can still browse the list of asset names (applications or metrics), but they cannot see query results, top tags, or facet details restricted by datasets. For instance, querying a metric with restricted data returns a blank graph, making it appear as if the query does not match any data.
```

Trying to remove the redundancy here with the word restricted in the same sentence :)

aliciascott commented on content/en/account_management/rbac/data_access.md:
```diff
@@ -0,0 +1,183 @@
+
+### APIs
+
+When querying data through Datadog APIs with restrictions enabled, users without permissions do not see query results that have been restricted by Restricted Datasets.
```
```suggestion
When querying data through Datadog APIs with restrictions enabled, users without permissions do **not** see query results that have been restricted by Restricted Datasets.
```

aliciascott reviewed (changes requested):
Hi @urseberry this looks good! I have some suggestions mostly for wording and flow and trying to be consistent with capitalizing or not capitalizing Restricted Dataset. Also I think the Supported and not supported examples could benefit from a little bit of re-formatting, but feel free to keep it as is!

urseberry commented:
I'm doing the changes in batches to make them easier to reason about. Will come back later to address the rest of the feedback.

urseberry commented:
Alicia, this is ready for your re-review. Some things to note:
- I accepted all your wording suggestions.
- I decided to make Restricted Dataset capitalized throughout. Thanks for pointing out the inconsistency!
- The in-app link to Data Access Control requires the `org_management` permission, which I don't have in any org, so I can't test it. I assume the PM gave me the right link. Do you have `org_management` anywhere?
- I adjusted your formatting for the examples a bit. I also removed the bolding from the field names, because I thought it made them hard to read. See attached screenshot. Thanks for the suggestion, though. I think the new version is clearer than what I originally had.
<img width="1041" alt="Screenshot 2024-11-27 at 12 45 25 AM" src="https://github.com/user-attachments/assets/1f722198-a2a1-4a10-9a62-93c015a835ea">

aliciascott commented on content/en/account_management/rbac/data_access.md:
```diff
@@ -0,0 +1,183 @@
+Log in as a user assigned the Datadog Admin role, or any user with a role in your organization with the [`user_access_manage` permission][5].
+
+1. Navigate to [Organization Settings][6].
+1. On the left side of the page, select [Data Access Controls][7].
```
Ah good point, I do not have that permission either

aliciascott reviewed (approved):
Looks good! Thanks for fixing those things, much easier to read and good flow :)

urseberry commented:
/merge
